### PR TITLE
Quit on failed login

### PR DESCRIPTION
**Pull request by DanielGelfand**
When a user fails to login, the bot should stop running instead of continuing to go to the target's page and attempting to get followers. In addition, there is now a popup when the user has bad login credentials.

### CODE REVIEW

File: src/GUI/bot_gui.py [hunk 1]
```diff
@@ -78,6 +80,7 @@ def __init__(self) -> None:
         self.root.config(bg=BACKGROUND_COLOR)
         self.targets_list = []
         self.is_running = False
+        self.bad_login = False
```
Comment (owenThurm) [inline src/GUI/bot_gui.py:83]: Is this variable necessary?

File: src/GUI/bot_gui.py [hunk 2]
```diff
@@ -189,6 +192,13 @@ def comment_round(self) -> None:
                 is_main,
                 send_to_brand_name=send_to_brand_name,
             )
+
+            if ig_bot.failed_login:
+                self.toggle_run_switch()
```
Comment (owenThurm) [inline src/GUI/bot_gui.py:197]: Might want to add `force_off=True` just to be sure.

File: src/instagram_bot.py [hunk 3]
```diff
@@ -57,17 +58,21 @@ def __init__(
...
+        if not self.failed_login:
+            sleep(1 + (random.random() * 3))
+            # Check for Notification popup
+            self.disband_notification()
+            sleep(1 + random.random())
+            # Goes the URL of target brand
+            self.get_brand()
+            sleep(1 + random.random())
+            # Click on list of followers
+            self.get_follower_list()
```
Comment (owenThurm) [inline src/instagram_bot.py:70]: Does it quit chromedriver if the login fails?

File: src/instagram_bot.py [hunk 4]
```diff
@@ -98,9 +103,16 @@ def login(self) -> None:
             except Exception as e:
                 print("Failed to land in homepage>>>")
                 print(e)
+                failed_login = True
 
         except Exception:
             print("Login failed....")
+            failed_login = True
```
Comment (owenThurm) [inline src/instagram_bot.py:110]: Can remove the `failed_login` var and just assigned `self.failed_login = True` in these Except blocks, thus allowing for my suggested change below

File: src/instagram_bot.py [hunk 5]
```diff
@@ -98,9 +103,16 @@ def login(self) -> None:
...
+
+        # If failed login, don't continue to search and scrape followers...
+        if failed_login:
+            self.failed_login = failed_login
+            self.quit()
```
Comment (owenThurm) [inline src/instagram_bot.py:115]:
```suggestion
        if self.failed_login:
            self.quit()
```

**Comment by owenThurm** [comment]
> When a user fails to login, the bot should stop running instead of continuing to go to the target's page and attempting to get followers. In addition, there is now a popup when the user has bad login credentials.

Good Idea 👍

**Comment by DanielGelfand** [comment]
> > When a user fails to login, the bot should stop running instead of continuing to go to the target's page and attempting to get followers. In addition, there is now a popup when the user has bad login credentials.
> 
> Good Idea 👍
failed_login

**Review by owenThurm — APPROVED** [review]
LGTM

**Pull request merged**